api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Client.java

```diff
@@ -65,15 +66,19 @@ class OAuth2Client implements OAuth2Authenticator, Closeable {
   private final Duration defaultAccessTokenLifespan;
   private final Duration defaultRefreshTokenLifespan;
   private final Duration refreshSafetyWindow;
+  private final Duration idleInterval;
   private final boolean tokenExchangeEnabled;
   private final HttpClient httpClient;
-  private final ScheduledExecutorService executor;
+  /* Visible for testing. */ final ScheduledExecutorService executor;
   private final boolean shouldCloseExecutor;
   private final ObjectMapper objectMapper;
   private final CompletableFuture<Void> started = new CompletableFuture<>();
+  /* Visible for testing. */ final AtomicBoolean sleeping = new AtomicBoolean();
+  private final Supplier<Instant> clock;
```

Comment:
Why's this a supplier instead of a `Clock`? (We do have that threeten-extra package which provides a mutable clock impl).

Reply:
Because it's more flexible than `Clock` and a `Clock` can be turned into a `Supplier<Instant>` with `clock::instant`.